lib/ansible/modules/cloud/docker/docker_network.py

```diff
@@ -253,29 +300,50 @@ def has_different_config(self, net):
                     if not (key in net['Options']) or value != net['Options'][key]:
                         different = True
                         differences.append('driver_options.%s' % key)
+
         if self.parameters.ipam_driver:
             if not net.get('IPAM') or net['IPAM']['Driver'] != self.parameters.ipam_driver:
                 different = True
                 differences.append('ipam_driver')
-        if self.parameters.ipam_options:
-            if not net.get('IPAM') or not net['IPAM'].get('Config'):
+
+        if self.parameters.ipam_config:
+            if not net.get('IPAM') or not net['IPAM']['Config']:
                 different = True
-                differences.append('ipam_options')
+                differences.append('ipam_config')
             else:
-                for key, value in self.parameters.ipam_options.items():
-                    camelkey = None
-                    for net_key in net['IPAM']['Config'][0]:
-                        if key == net_key.lower():
-                            camelkey = net_key
-                            break
-                    if not camelkey:
-                        # key not found
-                        different = True
-                        differences.append('ipam_options.%s' % key)
-                    elif net['IPAM']['Config'][0].get(camelkey) != value:
-                        # key has different value
-                        different = True
-                        differences.append('ipam_options.%s' % key)
+                for idx, ipam_config in enumerate(self.parameters.ipam_config):
+                    is_ipv4 = re.match(r'^([0-9]{1,3}\.){3}[0-9]{1,3}(/([0-9]|[1-2][0-9]|3[0-2]))?$',
+                                       ipam_config['subnet'])
+                    if is_ipv4:
+                        net_config = net['IPAM']['Config'][0]
+                    # assume ipv6 if not ipv4
+                    else:
+                        # ipv6 currently does not exist
+                        if len(net['IPAM']['Config']) < 2:
+                            net_config = dict()
+                        # ipv6 already exists
+                        else:
+                            net_config = net['IPAM']['Config'][1]
```

Comment:
Yes, IPv4 is always appended first, see https://github.com/moby/moby/blob/cf72051c3746fa60d7e7995c0342eaca91f07bfd/daemon/network.go#L675

Reply:
But there is a feature request for IPv6 only network (https://github.com/moby/moby/issues/37798). If this is implemented, the current logic would not work. Adjusting this code to work independently from index.

Reply:
It might be possible to convert the existing values to a syntax simliar to the one used by the module, and then use the comparison functions of #47709 to do the comparison. That would probably simplify the comparison logic a lot. But that's also something we can change later on (as long as we change it before 2.8 is released, afterwards we have to watch out for backwards compatibility).